File: add/codesnippet/CSharp/676b7f39-6ee0-49fa-b539-_2.cs
        // Asynchronous connect using host name (resolved by the 
        // BeginConnect call.)
        public static void BeginConnect3(string host, int port)
        {
            Socket s = new Socket(AddressFamily.InterNetwork,
                SocketType.Stream,
                ProtocolType.Tcp);

            allDone.Reset();

            Console.WriteLine("Establishing Connection to {0}", 
                host);
            s.BeginConnect(host, port, 
                new AsyncCallback(ConnectCallback1), s);

            // wait here until the connect finishes.  The callback 
            // sets allDone.
            allDone.WaitOne();

            Console.WriteLine("Connection established");
        }		

File: add/codesnippet/CPP/676b7f39-6ee0-49fa-b539-_2.cpp
   // Asynchronous connect using host name (resolved by the 
   // BeginConnect call.)
   static void BeginConnect3( String^ host, int port )
   {
      Socket^ s = gcnew Socket( AddressFamily::InterNetwork,SocketType::Stream,ProtocolType::Tcp );
      allDone->Reset();
      Console::WriteLine( "Establishing Connection to {0}", host );
      s->BeginConnect( host, port, gcnew AsyncCallback( ConnectCallback1 ), s );
      
      // wait here until the connect finishes.  The callback 
      // sets allDone.
      allDone->WaitOne();
      Console::WriteLine( "Connection established" );
   }

